src/snowflake/snowpark/mock/_udf_utils.py

```diff
@@ -11,6 +13,33 @@
 )
 from snowflake.snowpark.types import NullType, _NumericType
 
+VARIANT_INPUT_MAPPING = {
+    bytes: lambda x: x.decode("utf-8"),
+    Decimal: float,
+    date: str,
+    datetime: str,
+    time: str,
+    type(None): lambda _: SqlNullWrapper(),
+}
+
+
+class SqlNullWrapper:
+    def __init__(self) -> None:
+        self.is_sql_null = True
+
+
+def remove_null_wrapper(value):
+    if isinstance(value, SqlNullWrapper):
+        return None
+    return value
+
+
+def coerce_variant_input(value):
+    input_type = type(value)
+    if input_type in VARIANT_INPUT_MAPPING:
+        value = VARIANT_INPUT_MAPPING[input_type](value)
+    return value
```

Comment:
I'm curious what about other types like list/dict, they are kept as the original value?

Reply:
nvm I saw the test that `list` and `dict` are kept